aws/resource_aws_workspaces_workspace.go

```diff
@@ -171,7 +171,7 @@ func resourceAwsWorkspacesWorkspaceCreate(d *schema.ResourceData, meta interface
 
 	wsFail := resp.FailedRequests
 	if len(wsFail) > 0 {
-		return fmt.Errorf("workspace creation failed: %s", *wsFail[0].ErrorMessage)
+		return fmt.Errorf("workspace creation failed: %s: %s", *wsFail[0].ErrorCode, *wsFail[0].ErrorMessage)
```

Comment:
Good point 👍 AWS WorkSpaces service provides `FailedCreateWorkspaceRequest` and `FailedWorkspaceChangeRequest` data types only for workspace entity. Right now both of them are handled in one file, so I'd prefer to create the proposed abstraction once we get more `FailedXXXRequest`.

Reply:
I'm personally 👍  to also not creating these abstractions unless there is multiple usage -- there are a lot of service APIs that have similar "error code and message" fields within regular structure shapes since they are different than the shapes for modeled errors, so it could be a lot of abstracting for minimal benefit.